app/permissions_migrations.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import "github.com/mattermost/mattermost-server/model"
+
+type permissionTransformation struct {
+	On     func(map[string]bool) bool
+	Add    []string
+	Remove []string
+}
+type permissionsMap []permissionTransformation
+
+const (
+	MIGRATION_KEY_EMOJI_PERMISSIONS_SPLIT   = "emoji_permissions_split"
+	MIGRATION_KEY_WEBHOOK_PERMISSIONS_SPLIT = "webhook_permissions_split"
+
+	PERMISSION_MANAGE_EMOJIS                   = "manage_emojis"
+	PERMISSION_MANAGE_OTHERS_EMOJIS            = "manage_others_emojis"
+	PERMISSION_CREATE_EMOJIS                   = "create_emojis"
+	PERMISSION_DELETE_EMOJIS                   = "delete_emojis"
+	PERMISSION_DELETE_OTHERS_EMOJIS            = "delete_others_emojis"
+	PERMISSION_MANAGE_WEBHOOKS                 = "manage_webhooks"
+	PERMISSION_MANAGE_OTHERS_WEBHOOKS          = "manage_others_webhooks"
+	PERMISSION_MANAGE_INCOMING_WEBHOOKS        = "manage_incoming_webhooks"
+	PERMISSION_MANAGE_OTHERS_INCOMING_WEBHOOKS = "manage_others_incoming_webhooks"
+	PERMISSION_MANAGE_OUTGOING_WEBHOOKS        = "manage_outgoing_webhooks"
+	PERMISSION_MANAGE_OTHERS_OUTGOING_WEBHOOKS = "manage_others_outgoing_webhooks"
+)
```

Comment:
I could, but the reason to have this here is because in the future we can remove it from the model file, because is not longer needed, but will be always needed in the migrations itself.

Reply:
If you writting a plugin that needs to deal with both permission you need to be explicit about that. I think is not an easy task, and actually, I can keep that in model, but then, you still need to detect the version and use the previous permission or the new permission depending on your version. We can keep the old permission in model during some time, but I'm not 100% sure about that.

Reply:
Makes sense. I think we can ignore this particular concern for now given the current stability of our plugin APIs.